target_postgres/tests/test_standard_target.py

```diff
@@ -232,3 +244,94 @@ def test_new_array_column(postgres_target):
+    postgres_config_hard_delete_true["hard_delete"] = True
+    pg_hard_delete_true = TargetPostgres(config=postgres_config_hard_delete_true)
+    singer_file_to_target(file_name, pg_hard_delete_true)
+    engine = sqlalchemy_engine(postgres_config)
+    with engine.connect() as connection:
+        result = connection.execute("SELECT * FROM test_activate_version_hard")
+        assert result.rowcount == 7
+        # Add a record like someone would if they weren't using the tap target combo
+        result = connection.execute(
+            "INSERT INTO test_activate_version_hard(code, \"name\") VALUES('Manual1', 'Meltano')"
+        )
+        result = connection.execute(
+            "INSERT INTO test_activate_version_hard(code, \"name\") VALUES('Manual2', 'Meltano')"
+        )
+        result = connection.execute("SELECT * FROM test_activate_version_hard")
+        assert result.rowcount == 9
+
+    singer_file_to_target(file_name, pg_hard_delete_true)
+
+    # Should remove the 2 records we added manually
+    with engine.connect() as connection:
+        result = connection.execute("SELECT * FROM test_activate_version_hard")
+        assert result.rowcount == 7
+
+
+def test_activate_version_soft_delete(postgres_config):
+    """Activate Version Soft Delete Test"""
+    file_name = "activate_version_soft.singer"
+    engine = sqlalchemy_engine(postgres_config)
+    with engine.connect() as connection:
+        result = connection.execute("DROP TABLE IF EXISTS test_activate_version_soft")
+    postgres_config_soft_delete = copy.deepcopy(postgres_config)
+    postgres_config_soft_delete["hard_delete"] = False
+    pg_soft_delete = TargetPostgres(config=postgres_config_soft_delete)
+    singer_file_to_target(file_name, pg_soft_delete)
+
+    with engine.connect() as connection:
+        result = connection.execute("SELECT * FROM test_activate_version_soft")
+        assert result.rowcount == 7
+        # Add a record like someone would if they weren't using the tap target combo
+        result = connection.execute(
+            "INSERT INTO test_activate_version_soft(code, \"name\") VALUES('Manual1', 'Meltano')"
+        )
+        result = connection.execute(
+            "INSERT INTO test_activate_version_soft(code, \"name\") VALUES('Manual2', 'Meltano')"
+        )
+        result = connection.execute("SELECT * FROM test_activate_version_soft")
+        assert result.rowcount == 9
+
+    singer_file_to_target(file_name, pg_soft_delete)
+
+    # Should have all records including the 2 we added manually
+    with engine.connect() as connection:
+        result = connection.execute("SELECT * FROM test_activate_version_soft")
+        assert result.rowcount == 9
+
+        result = connection.execute(
+            "SELECT * FROM test_activate_version_soft where _sdc_deleted_at is NOT NULL"
+        )
+        assert result.rowcount == 2
+
+
+def test_activate_version_doesnt_delete_before_populating(postgres_config):
```

Comment:
This fails right now re https://github.com/MeltanoLabs/target-postgres/pull/89/files#r1084618420

Reply:
This may be desired behavior, https://github.com/transferwise/pipelinewise-tap-postgres/blob/b78ba51f3ff239c8de762f6eaa5406f53ec7b964/tap_postgres/sync_strategies/full_table.py 

I think the idea is if it's the first run you want to drop the table first. Hmmmm seems like an odd implementation but it doesn't look like a spec issue honestly.